### PR TITLE
omit empty-label fields

### PR DESCRIPTION
**Pull request by amery**

### CODE REVIEW

File: internal/fields.go [hunk 1]
```diff
@@ -14,3 +14,37 @@ func SortedKeys(fields map[string]any) []string {
...
+		}
+
+		fields = m
+		count = len(m)
+	}
+
+	return fields, count > 0
+}
+
+func hasEmptyLabels(fields map[string]any) bool {
```
Comment (karasz) [inline internal/fields.go:42]:
```go
func hasEmptyLabels(fields map[string]any) bool {
    _,ok:=fileds[""]
    return ok
}

File: internal/fields.go [hunk 2]
```diff
@@ -14,3 +14,37 @@ func SortedKeys(fields map[string]any) []string {
...
+
+// FilterEmptyFields removes fields with empty labels
+func FilterEmptyFields(fields map[string]any) (map[string]any, bool) {
+	count := len(fields)
+
+	if count == 0 {
+		return fields, false
+	}
+
+	if hasEmptyLabels(fields) {
```
Comment (karasz) [inline internal/fields.go:26]:
```go
delete(fields,"")
```
the rest is not needed.

**Comment by amery** [comment]
superfluous `FilterEmptyFields()` helper removed

File: internal/fields.go [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package internal
+
+import "sort"
+
+// SortedKeys returns a sorted list of non-empty field keys
+func SortedKeys(fields map[string]any) []string {
```
Comment (karasz) [inline internal/fields.go:6]:
```go
func SortedKeys(fields map[string]any) []string {
    keys := make([]string, 0,len(fields))
    delete(fields, "")
    for key := range fields {
        keys = append(keys,key)
    }
    sort.Strings(keys)
    return keys
}
```

**Review by karasz — CHANGES_REQUESTED** [review]
See comment

File: internal/fields.go (hunk 3 above)
Comment (amery) [reply internal/fields.go:6]: are you sure the delete is cheaper than the check?

**Comment by amery** [comment]
@karasz see my response

File: internal/fields.go (hunk 3 above)
Comment (karasz) [reply internal/fields.go:6]: Probably the same for normal sized maps, and I don't think it will have to manage ones with many million keys. GTG

**Review by karasz — APPROVED** [review]

**Comment by amery** [comment]
`v0.4.2` tagged. also +0.0.1 on each affected handler

**Pull request merged**